### PR TITLE
Fix #345

### LINKED ISSUE #345: Run the TestInit.test_nofilematch test in a tempdir
The current test is
```
    def test_nofilematch(self):
        with pytest.raises(ValueError, match="No file has been matched to"):
>           ABFE(
                dir="./",
                prefix="dhdl",
                suffix="xvg",
                T=310,
            )
```
Which is fine it's own sense but will try to search through all the subfolders in dir="./". So if you run it in your root path '~', it will try to access all the files which might annoy some security people. Once I had time should make a temp dir for this test.

### PR DESCRIPTION
**Pull request by xiki-tempula**
Fix #345

### CODE REVIEW

File: devtools/conda-envs/test_env.yaml [hunk 1]
```diff
@@ -24,5 +24,5 @@ dependencies:
 - sphinx_rtd_theme
 
 - pip:
-  - alchemtest @ https://github.com/alchemistry/alchemtest/archive/master.zip
+  - alchemtest @ git+https://github.com/xiki-tempula/alchemtest.git@feat_main
```
Comment (orbeckst) [inline devtools/conda-envs/test_env.yaml:27]: change back

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -50,7 +50,7 @@ jobs:
       with:
         environment-file: devtools/conda-envs/test_env.yaml
         environment-name: test
-        extra-specs: |
+        create-args: |
```
Comment (orbeckst) [inline .github/workflows/ci.yaml:53]: Is this also being fixed in #346 ?

**Review by orbeckst — CHANGES_REQUESTED** [review]
LGTM, just fix the CI that pulls from your personal GH.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (xiki-tempula) [reply .github/workflows/ci.yaml:53]: Yes.

**Comment by xiki-tempula** [comment]
Remove the py3.12 support now. I will add the py3.12 support in another PR, where I think I will update the versioneer. I would hope to do the same some of updating to the cookie cutter here but I guess that will generate big merge conflict for https://github.com/alchemistry/alchemlyb/pull/348

File: docs/requirements.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+name: docs
+channels:
+dependencies:
+    # Base depends
+  - python=3.10
```
Comment (xiki-tempula) [inline docs/requirements.yaml:5]: Pin it to a python version so it would not break just due to some python update.

**Review by orbeckst — APPROVED** [review]

**Pull request merged**